app/models/package.rb

```diff
@@ -602,6 +608,30 @@ def donor_condition_name
 
   private
 
+  def create_and_save_copy
+    copy = dup
+    copy.quantity = 1
+    copy.received_quantity = 1
+    copy.inventory_number = generate_q_inventory_number
+    copy_and_save_images(copy) if copy.save
+  end
+
+  def generate_q_inventory_number
+    existing_q_inventory_number_count = Package.where("inventory_number LIKE ?", "#{inventory_number}Q%").count
```

Comment:
ordinarily this would be fine but in Stockit, we allow people to enter Q inventory numbers manually. It's possible they could add a number that isn't in sequential order and therefore we would get a naming clash.

Example:
F00001Q1
F00001Q3

If we now generate an inventory number here then it will try to make F00001Q3.

Solution: I think it would be fine to list inventory Q numbers, sorted in desc order, limit 1 and increment.

```
largest_q_number = Package.where("inventory_number LIKE ?", "#{inventory_number}Q%").order(desc).limit(1)
return largest_q_number.split('Q').last.to_i + 1
```